airflow/contrib/hooks/salesforce_hook.py

```diff
@@ -0,0 +1,471 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module contains a Salesforce Hook
+which allows you to connect to your Salesforce instance,
+retrieve data from it, and write that data to a file
+for other uses.
+
+NOTE:   this hook also relies on the simple_salesforce package:
+        https://github.com/simple-salesforce/simple-salesforce
+"""
+from simple_salesforce import Salesforce
+from airflow.hooks.base_hook import BaseHook
+
+import logging
+import json
+
+import pandas as pd
+import time
+
+
+class SalesforceHook(BaseHook):
+    def __init__(
+            self,
+            conn_id,
+            *args,
+            **kwargs
+    ):
+        """
+        Create new connection to Salesforce
+        and allows you to pull data out of SFDC and save it to a file.
+
+        You can then use that file with other
+        Airflow operators to move the data into another data source
+
+        :param conn_id:     the name of the connection that has the parameters
+                            we need to connect to Salesforce.
+                            The conenction shoud be type `http` and include a
+                            user's security token in the `Extras` field.
+        .. note::
+            For the HTTP connection type, you can include a
+            JSON structure in the `Extras` field.
+            We need a user's security token to connect to Salesforce.
+            So we define it in the `Extras` field as:
+                `{"security_token":"YOUR_SECRUITY_TOKEN"}`
+        """
+        self.conn_id = conn_id
+        self._args = args
+        self._kwargs = kwargs
+
+        # get the connection parameters
+        self.connection = self.get_connection(conn_id)
+        self.extras = self.connection.extra_dejson
+
+        # define a map of strings to different schema converting functions
+        self.SF_TO_SCHEMA_MAP = {
+            "BQ": self.convertSalesforceSchemaToBQ
+        }
+
+    def signIn(self):
+        """
+        Sign into Salesforce.
+
+        If we have already signed it, this will just return the original object
+        """
+        if hasattr(self, 'sf'):
+            return self.sf
+
+        # connect to Salesforce
+        sf = Salesforce(
+            username=self.connection.login,
+            password=self.connection.password,
+            security_token=self.extras['security_token'],
+            instance_url=self.connection.host
+        )
+        self.sf = sf
+        return sf
+
+    def makeQuery(self, query):
+        """
+        Make a query to Salesforce.  Returns result in dictionary
+
+        :param query:    The query to make to Salesforce
+        """
+        if not hasattr(self, 'sf'):
```

Comment:
You're right.  That one is my bad.  I forgot to go and update all of these after I created the sign_in function.